### PR TITLE
JWWEB-730 Create wallet process refactoring

### PR DESCRIPTION
**Pull request by IViol**

### CODE REVIEW

File: src/components/WalletBackupForm/WalletBackupForm.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ import {
 import walletBackupFormStyle from './walletBackupForm.m.scss'
 
 type Props = {|
-  +handleSubmit: (?SyntheticEvent<HTMLFormElement>) => Promise<?FormFields>,
+  +handleSubmit: (?SyntheticEvent<HTMLFormElement>) => ?Promise<?FormFields>,
```
Comment (IViol) [inline src/components/WalletBackupForm/WalletBackupForm.js]: fixed because of `flow`

Comment (IViol) [reply src/components/WalletBackupForm/WalletBackupForm.js]: final-form handleSubmit has optional promise type for return value

File: src/pages/WalletsCreate/WalletsCreateView.js [hunk 2]
```diff
@@ -1,122 +1,319 @@
...
   componentDidMount() {
-    this.props.openView()
+    this.props.requestBlockNumbers()
   }
 
-  componentWillUnmount() {
-    this.props.closeView()
+  setCurrentStep = (currentStep: WalletsCreateStep) => () => {
+    this.setState({ currentStep })
+    gaSendEvent('CreateWallet', EVENTS[currentStep])
```
Comment (IViol) [inline src/pages/WalletsCreate/WalletsCreateView.js]: need to fix

File: src/components/WalletBackupForm/WalletBackupForm.js [hunk 3]
```diff
@@ -75,13 +76,12 @@ export function WalletBackupForm({
...
       <h2 className={walletBackupFormStyle.title}>{t`Back Up ${name}`}</h2>
       <p className={walletBackupFormStyle.text}>
-        {hasMnemonicAttributes
-          ? getMnemonicAttributesFirstLine(passphrase, derivationPath)
-          : BACKUP_KEY_FIRST_LINE
-        }
-      </p>
-      <p className={walletBackupFormStyle.text}>
-        {hasMnemonicAttributes ? BACKUP_MNEMONIC_SECOND_LINE : BACKUP_KEY_SECOND_LINE}
+        {backupText.map((text: string, index: number) => (
```
Comment (stcherenkov) [inline src/components/WalletBackupForm/WalletBackupForm.js]: Maybe `.split('\n').join('<br />')` and `dangerouslySetInnerHTML` would look simpler?

File: src/components/WalletBackupForm/WalletBackupForm.js [hunk 4]
```diff
@@ -113,7 +113,7 @@ export function WalletBackupForm({
           type='button'
           theme='general'
           className={walletBackupFormStyle.button}
-          onClick={handleDownload}
+          onClick={handleDownload(values.data, passphrase, derivationPath)}
```
Comment (stcherenkov) [inline src/components/WalletBackupForm/WalletBackupForm.js]: Bad practice, you create new function on every props check resulting in rerender.

You need to memoize the resulting function or make it method of class instance.

Comment (IViol) [reply src/components/WalletBackupForm/WalletBackupForm.js]: we always have just one set of these props

Comment (stcherenkov) [reply src/components/WalletBackupForm/WalletBackupForm.js]: It does not matter. You still create a new function. And react always calls render (or function component body in your case) and then makes strict comparison of props by reference – and you will always have a new reference here.

Try in browser console, for example:
```js
function foo(i, j) {
  return () => i + j
}

console.log(foo(0, 1) === foo(0, 1))
```

Arguments are the same, but results are not equal.

Comment (IViol) [reply src/components/WalletBackupForm/WalletBackupForm.js]: Sure, I know it.
But especially for this component it absolutely is not a problem.

Comment (stcherenkov) [reply src/components/WalletBackupForm/WalletBackupForm.js]: True.

**Pull request merged**